pennylane_lightning/core/src/simulators/lightning_gpu/catalyst/LightningGPUSimulator.cpp

```diff
@@ -0,0 +1,568 @@
+               "Invalid key for cached observables");
+
+    // update tape caching
+    if (this->tape_recording) {
+        cache_manager.addObservable(obsKey, MeasurementsT::Expval);
+    }
+
+    auto &&obs = this->obs_manager.getObservable(obsKey);
+
+    Pennylane::LightningGPU::Measures::Measurements<StateVectorT> m{
+        *(this->device_sv)};
+
+    return device_shots ? m.expval(*obs, device_shots, {}) : m.expval(*obs);
+}
+
+auto LightningGPUSimulator::Var(ObsIdType obsKey) -> double {
+    RT_FAIL_IF(!this->obs_manager.isValidObservables({obsKey}),
+               "Invalid key for cached observables");
+
+    // update tape caching
+    if (this->tape_recording) {
+        this->cache_manager.addObservable(obsKey, MeasurementsT::Var);
+    }
+
+    auto &&obs = this->obs_manager.getObservable(obsKey);
+
+    Pennylane::LightningGPU::Measures::Measurements<StateVectorT> m{
+        *(this->device_sv)};
+
+    return device_shots ? m.var(*obs, device_shots) : m.var(*obs);
+}
+
+void LightningGPUSimulator::State(DataView<std::complex<double>, 1> &state) {
+    const std::size_t num_qubits = this->device_sv->getNumQubits();
+    const std::size_t size = Pennylane::Util::exp2(num_qubits);
+    RT_FAIL_IF(state.size() != size,
+               "Invalid size for the pre-allocated state vector");
+
+    // create a temporary buffer to copy the underlying state-vector to
+    std::vector<std::complex<double>> buffer(size);
+    // copy data from device to host
+    this->device_sv->CopyGpuDataToHost(buffer.data(), size);
+
+    // move data to state leveraging MemRefIter
+    std::move(buffer.begin(), buffer.end(), state.begin());
+}
+
+void LightningGPUSimulator::Probs(DataView<double, 1> &probs) {
+    Pennylane::LightningGPU::Measures::Measurements<StateVectorT> m{
+        *(this->device_sv)};
+    auto &&dv_probs = device_shots ? m.probs(device_shots) : m.probs();
+
+    RT_FAIL_IF(probs.size() != dv_probs.size(),
+               "Invalid size for the pre-allocated probabilities");
+
+    std::move(dv_probs.begin(), dv_probs.end(), probs.begin());
+}
+
+void LightningGPUSimulator::PartialProbs(
+    DataView<double, 1> &probs, const std::vector<QubitIdType> &wires) {
+    const std::size_t numWires = wires.size();
+    const std::size_t numQubits = this->GetNumQubits();
+
+    RT_FAIL_IF(numWires > numQubits, "Invalid number of wires");
+    RT_FAIL_IF(!isValidQubits(wires), "Invalid given wires to measure");
+
+    auto dev_wires = getDeviceWires(wires);
+    Pennylane::LightningGPU::Measures::Measurements<StateVectorT> m{
+        *(this->device_sv)};
+    auto &&dv_probs =
+        device_shots ? m.probs(dev_wires, device_shots) : m.probs(dev_wires);
+
+    RT_FAIL_IF(probs.size() != dv_probs.size(),
+               "Invalid size for the pre-allocated partial-probabilities");
+
+    std::move(dv_probs.begin(), dv_probs.end(), probs.begin());
+}
+
+void LightningGPUSimulator::Sample(DataView<double, 2> &samples,
```

Comment:
Now that #927 is merged, I think lightning.gpu can also support seeding on its `generate_samples`. See https://github.com/PennyLaneAI/pennylane-lightning/pull/927/files#diff-879c9c8443c0af6a40f50c3b27d8140bbb41ce92e3626fd005870a9e85accb6e for the treatment.

Reply:
Done!